### PR TITLE
Scheduler:remove direct dependency for k8s.io/kubernetes/pkg/util/node and  remove // TODO: prefer stable labels for zone in v1.18

### PR DESCRIPTION
**Pull request by tanjunchen**
**What type of PR is this?**

 /kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

ref:https://github.com/kubernetes/kubernetes/issues/89930

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
None
```

### CODE REVIEW

**Review by damemi — APPROVED** [review]
this lgtm, but it would be nice in the future if we could find a way to have this function in a single external package that can be used by both the scheduler and nodelifecycle controller (that seems to be the only other place this function is used)

/lgtm

File: pkg/scheduler/util/utils.go [hunk 1]
```diff
@@ -109,3 +109,39 @@ func GetPodAntiAffinityTerms(podAntiAffinity *v1.PodAntiAffinity) (terms []v1.Po
...
+// GetZoneKey will first check failure-domain.beta.kubernetes.io/zone and if not exists, will then check
+// topology.kubernetes.io/zone
+// NOTE: This function copied from pkg/util/node package to avoid external scheduler dependency
+func GetZoneKey(node *v1.Node) string {
+	labels := node.Labels
+	if labels == nil {
+		return ""
+	}
+
+	// TODO: prefer stable labels for zone in v1.18
```
Comment (liggitt) [inline pkg/scheduler/util/utils.go:127]: can just look at the stable labels. a 1.19 scheduler can depend on a 1.18+ controller, which ensures the stable labels exist and are synced with the beta labels.

File: pkg/scheduler/util/utils.go [hunk 2]
```diff
@@ -109,3 +109,39 @@ func GetPodAntiAffinityTerms(podAntiAffinity *v1.PodAntiAffinity) (terms []v1.Po
...
+	zone, ok := labels[v1.LabelZoneFailureDomain]
+	if !ok {
+		zone, _ = labels[v1.LabelZoneFailureDomainStable]
+	}
+
+	// TODO: prefer stable labels for region in v1.18
+	region, ok := labels[v1.LabelZoneRegion]
+	if !ok {
+		region, _ = labels[v1.LabelZoneRegionStable]
+	}
```
Comment (liggitt) [inline pkg/scheduler/util/utils.go:137]:
```suggestion
	zone := labels[v1.LabelZoneFailureDomainStable]
	region := labels[v1.LabelZoneRegionStable]
```

**Comment by liggitt** [comment]
/lgtm

scheduling team has approval

**Comment by liggitt** [comment]
looks like unit tests need updating to stable labels

**Review by gavinfish — COMMENTED** [review]
/lgtm

**Comment by tanjunchen** [comment]
> looks like unit tests need updating to stable labels

yes . i have updated it , could you review  it again?

**Comment by liggitt** [comment]
lgtm
/unassign
/assign @ahg-g

File: pkg/scheduler/util/utils.go [hunk 3]
```diff
@@ -109,3 +109,25 @@ func GetPodAntiAffinityTerms(podAntiAffinity *v1.PodAntiAffinity) (terms []v1.Po
 	}
 	return terms
 }
+
```
Comment (ahg-g) [inline pkg/scheduler/util/utils.go:112]: defaultpodtopologyspread will be deprecated soon, so add a todo to move this function to node_tree.go once this happens.

**Comment by ahg-g** [comment]
On a second thought, there is no urgency to doing this, our plan is to deprecate the util pkg and so we shouldn't add anything to it. Also, what we should invest in is moving k8s.io/kubernetes/pkg/util/node pkg to staging instead of duplicating the code.

File: pkg/scheduler/util/utils.go (hunk 3 above)
Comment (tanjunchen) [reply pkg/scheduler/util/utils.go:112]: sure

Comment (tanjunchen) [reply pkg/scheduler/util/utils.go:112]: @ahg-g 
updated

**Pull request closed without merging**